servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpObjectDecoder.java

```diff
@@ -544,40 +544,42 @@ private void parseHeaderLine(HttpHeaders headers, ByteBuf buffer, final int lfIn
         //                     ; obsolete line folding
         //                     ; see Section 3.2.4
         final int nonControlIndex = lfIndex - 2;
-        int headerStart = buffer.forEachByte(buffer.readerIndex(), nonControlIndex - buffer.readerIndex(),
+        final int headerStart = buffer.forEachByte(buffer.readerIndex(), nonControlIndex - buffer.readerIndex(),
                 FIND_NON_LINEAR_WHITESPACE);
         if (headerStart < 0) {
             throw new IllegalArgumentException("unable to find start of header name");
         }
 
-        int headerEnd = buffer.forEachByte(headerStart + 1, nonControlIndex - headerStart,
+        final int headerEnd = buffer.forEachByte(headerStart + 1, nonControlIndex - headerStart,
                 FIND_COLON_OR_WHITE_SPACE);
         if (headerEnd < 0) {
             throw new IllegalArgumentException("unable to find end of header name");
         }
+        // We assume the allocator will not leak memory, and so we retain + slice to avoid copying data.
+        final CharSequence name = newAsciiString(newBufferFrom(
+                buffer.retainedSlice(headerStart, headerEnd - headerStart)));
 
         int valueStart = headerEnd + 1;
-        // We assume the allocator will not leak memory, and so we retain + slice to avoid copying data.
-        CharSequence name = newAsciiString(newBufferFrom(buffer.retainedSlice(headerStart, headerEnd - headerStart)));
         if (buffer.getByte(headerEnd) != COLON_BYTE) {
-            valueStart = buffer.forEachByte(headerEnd + 1, nonControlIndex - headerEnd, FIND_COLON);
+            valueStart = buffer.forEachByte(headerEnd + 1, nonControlIndex - headerEnd, FIND_COLON) + 1;
             if (valueStart < 0) {
                 throw new IllegalArgumentException("unable to find colon");
             }
         }
         if (nonControlIndex <= valueStart) {
             headers.add(name, emptyAsciiString());
         } else {
-            valueStart = buffer.forEachByte(valueStart + 1, nonControlIndex - valueStart, FIND_NON_LINEAR_WHITESPACE);
+            valueStart = buffer.forEachByte(valueStart, nonControlIndex - valueStart + 1, FIND_NON_LINEAR_WHITESPACE);
             // Find End Of String
-            int valueEnd;
+            final int valueEnd;
             if (valueStart < 0 || (valueEnd = buffer.forEachByteDesc(valueStart, lfIndex - valueStart - 1,
                     FIND_NON_LINEAR_WHITESPACE)) < 0) {
                 headers.add(name, emptyAsciiString());
             } else {
                 // We assume the allocator will not leak memory, and so we retain + slice to avoid copying data.
-                headers.add(name, newAsciiString(newBufferFrom(
-                        buffer.retainedSlice(valueStart, valueEnd - valueStart + 1))));
+                final CharSequence value = newAsciiString(newBufferFrom(
```

Comment:
IIUC this change is formatting related ... can you revert and keep the inlined approach as before?